### PR TITLE
haproxy setup for couchdb2

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: src/commcare_cloud/ansible/roles/haproxy/templates/service.j2 [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+{% set _ = entry.append(item.service.haproxy_backend_nodes | count | string) %}
+    {{ entry | join(' ') }}
+{% endfor %}
+
+{% for host_name in item.service.haproxy_backup_nodes|default([]) %}
+{% if hostvars[host_name] is defined %}
+{%   set ip_addr = hostvars[host_name].internal_ipv4.address %}
+{% endif %}
+{% set entry = [] %}
+{% set _ = entry.append("server") %}
```
Comment (millerdev) [inline src/commcare_cloud/ansible/roles/haproxy/templates/service.j2]: Interesting use of mutating side-effects. Could it be structured like this instead?
```
{% set entry = [
    "server",
    ((host_name.name | default(host_name)) | string),
    ...
] %}
{{ entry | join(' ') }}
```
or even better
```
{{ [
    "server",
    ((host_name.name | default(host_name)) | string),
    ...
] | join(' ') }}
```

File: src/commcare_cloud/ansible/roles/thp/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+- stat:
+    path: /etc/default/grub
+    register: uses_grub
+
+- stat:
+    path: /sys/kernel/mm/transparent_hugepage/enabled
+  register: thp_enabled
```
Comment (millerdev) [inline src/commcare_cloud/ansible/roles/thp/tasks/main.yml]: Inconsistent indentation (also below)

File: src/commcare_cloud/ansible/roles/haproxy/templates/service.j2 (hunk 1 above)
Comment (snopoke) [reply src/commcare_cloud/ansible/roles/haproxy/templates/service.j2]: interesting, that would read better. Most of this role is copied from https://github.com/openstack/openstack-ansible-haproxy_server as noted in the README.

I'll test this out and update.

**Comment by snopoke** [comment]
rolled this out to ICDS

**Pull request merged**